test/mbedclient/utest/m2mnsdlinterface/test_m2mnsdlinterface.cpp

```diff
@@ -1616,40 +1646,38 @@ void Test_M2MNsdlInterface::test_find_resource()
                                             "resource_name");
 
     M2MResourceInstance *resource_instance = new M2MResourceInstance(*resource,
-                                                                     "name",
+                                                                     "0",
                                                                      "resource_type",
-                                                                     M2MResourceInstance::INTEGER,0,"",false);
+                                                                     M2MResourceInstance::INTEGER,0,"0",false);
 
     m2mobjectinstance_stub::base_type = M2MBase::ObjectInstance;
     m2mobject_stub::instance_list.push_back(object_instance);
     m2mobjectinstance_stub::resource_list.push_back(resource);
     m2mresource_stub::list.push_back(resource_instance);
     m2mobject_stub::base_type = M2MBase::Object;
     nsdl->_object_list.push_back(object);
-
+    m2mresource_stub::bool_value = true;
     CHECK(nsdl->find_resource("name") != NULL);
+    CHECK(nsdl->find_resource("name1") == NULL);
     CHECK(nsdl->find_resource("name", token, 4) != NULL);
     CHECK(nsdl->find_resource("name", token, 5) == NULL);
 
-    m2mbase_stub::int_value  = 0;
-    m2mobject_stub::base_type = M2MBase::ObjectInstance;
+    m2mbase_stub::ret_counter = 0;
     CHECK(nsdl->find_resource("name/0") != NULL);
+    m2mbase_stub::ret_counter = 0;
+    CHECK(nsdl->find_resource("name/0", inst_token, 4) != NULL);
+    m2mbase_stub::ret_counter = 0;
```

Comment:
Would it be worth the trouble to test also that returned value is correct? Eg. "CHECK(nsdl->find_resource("name/0", inst_token, 4) != NULL);" could be replaced with "CHECK(nsdl->find_resource("name/0", inst_token, 4) == object_instance);".